PlagiarismPlugin.inc.php

```diff
@@ -43,215 +106,905 @@ public function getDescription() {
+			// not going to sent it for plagiarism check if EULA not stamped to submission or submitter
+			if (!$submission->getData('ithenticate_eula_version') || !$user->getData('ithenticateEulaVersion')) {
+				$this->sendErrorMessage('Unable to obtain the stamped EULA details to submission or submitter', $submission->getId());
+				return false;
+			}
+		}
+
+		/** @var DAOResultIterator $submissionFiles */
+		$submissionFiles = Services::get('submissionFile')->getMany([
+            'submissionIds' => [$submission->getId()],
+		]);
+
+		try {
+			foreach($submissionFiles as $submissionFile) { /** @var SubmissionFile $submissionFile */
+				if (!$this->createNewSubmission($request, $user, $submission, $submissionFile, $ithenticate)) {
+					return false;
+				}
+			}
+		} catch (\Throwable $exception) {
+			$this->sendErrorMessage($exception->getMessage(), $submission->getId());
+			return false;
 		}
-		error_log('iThenticate submission '.$submissionid.' failed: '.$message);
+
+		$submission->setData('ithenticate_submission_completed_at', Core::getCurrentDate());
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /** @var SubmissionDAO $submissionDao */
+		$submissionDao->updateObject($submission);
+
+		return false;
 	}
 
 	/**
-	 * Send submission files to iThenticate.
-	 * @param $hookName string
-	 * @param $args array
+	 * Add ithenticate related data and actions to submission file grid view
+	 * 
+	 * @param string $hookName `editorsubmissiondetailsfilesgridhandler::initfeatures` or `editorreviewfilesgridhandler::initfeatures`
+	 * @param array $params
+	 * 
+	 * @return bool
 	 */
-	public function callback($hookName, $args) {
+	public function addActionsToSubmissionFileGrid($hookName, $params) {
 		$request = Application::get()->getRequest();
+		$user = $request->getUser();
 		$context = $request->getContext();
-		$contextPath = $context->getPath();
-		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
-		$submission = $submissionDao->getById($request->getUserVar('submissionId'));
-		$publication = $submission->getCurrentPublication();
 
-		require_once(dirname(__FILE__) . '/vendor/autoload.php');
+		if (!$user->hasRole([ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_REVIEWER], $context->getId())) {
+			return false;
+		}
 
-		// try to get credentials for current context otherwise use default config
-        	$contextId = $context->getId();
-		list($username, $password) = $this->getForcedCredentials();
-		if (empty($username) || empty($password)) {
-			$username = $this->getSetting($contextId, 'ithenticateUser');
-			$password = $this->getSetting($contextId, 'ithenticatePass');
+		/** @var EditorSubmissionDetailsFilesGridHandler|EditorReviewFilesGridHandler $submissionDetailsFilesGridHandler */
+		$submissionDetailsFilesGridHandler = & $params[0];
+
+		import('plugins.generic.plagiarism.grids.SimilarityActionGridColumn');
```

Comment:
Use `$this->import` on the plugin object to import code inside the plugin's installation directory.